gui/src/main.rs

```diff
@@ -0,0 +1,146 @@
+
+    let mut editor = Editor::new();
+    let mut fps_ctrl = FPSControl::new().with_max_fps(60);
+
+    if args.testing {
+        editor.instant = true;
+        editor.fixed_seed = true;
+        editor.auto_generate = true;
+        editor.edit_gen_config = true;
+    }
+
+    if let Some(config_name) = args.config {
+        if editor.gen_configs.contains_key(&config_name) {
+            editor.current_gen_config = config_name;
+        }
+    }
+
+    loop {
+        fps_ctrl.on_frame_start();
+        editor.on_frame_start();
+
+        // optionally, start generating next map right away
+        if editor.is_paused() && editor.auto_generate {
+            editor.set_playing();
+        }
+
+        // perform walker step
+        let steps = match editor.instant {
+            true => usize::max_value(),
+            false => editor.steps_per_frame,
+        };
+
+        if editor.gen.as_ref().is_some() {
+            for _ in 0..steps {
+                if editor.is_paused() || editor.gen.as_ref().unwrap().walker.finished {
+                    break;
+                }
+
+                editor.gen.as_mut().unwrap().step().unwrap_or_else(|err| {
+                    println!("Walker Step Failed: {:}", err);
+                    editor.set_setup();
+                });
+
+                // walker did a step using SingleStep -> now pause
+                if editor.is_single_setp() {
+                    editor.set_stopped();
+                }
+            }
+
+            // this is called ONCE after map was generated
+            if editor.gen.as_ref().unwrap().walker.finished && !editor.is_setup() {
+                // kinda crappy, but ensure that even a panic doesnt crash the program
+                let _ = panic::catch_unwind(AssertUnwindSafe(|| {
+                    editor
+                        .gen
+                        .as_mut()
+                        .unwrap()
+                        .post_processing()
+                        .unwrap_or_else(|err| {
+                            println!("Post Processing Failed: {:}", err);
+                        });
+                }));
+
+                // switch into setup mode for next map
+                editor.set_setup();
+            }
+
+            editor.define_egui();
+            editor.set_cam();
+            editor.handle_user_inputs();
+
+            clear_background(WHITE);
+
+            // draw_grid_blocks(&editor.gen.map.grid);
+            draw_chunked_grid(
+                &editor.gen.as_ref().unwrap().map.grid,
+                &editor.gen.as_ref().unwrap().map.chunks_edited,
+                editor.gen.as_ref().unwrap().map.chunk_size,
+            );
+            draw_walker_kernel(&editor.gen.as_ref().unwrap().walker, KernelType::Outer);
+            draw_walker_kernel(&editor.gen.as_ref().unwrap().walker, KernelType::Inner);
+            draw_walker(&editor.gen.as_ref().unwrap().walker);
+        } else {
+            continue;
+        }
```

Comment:
this skips the entire drawing if gen is not set, which it isnt on program startup. this results in gui not loading

Reply:
I just removed this continue block, _should_ fix this issue